### ISSUE TITLE
Webinterface Backup Export/Import

### ISSUE THREAD

**Issue by therealschimmi** [TandoorRecipes/recipes#79]
Can you include some option to export all recipes or even everything into one zip file?

**Comment by vabene1111** [comment]
in what format would you like such a feature ? 

right now you can easily export the whole database using `pg_dump` or alternativley by creating django fixtures `python manage.py dumpdata cookbook`. Both of these options are nice to create backups but are probably of not much use if you want to import into any other kind of recipe manager.

**Comment by therealschimmi** [comment]
I dont care about compatibility too much. I would just like to have a backup in case i screw up something with Docker and lose all content. Just make it accessible from the webinterface :)

**Comment by vabene1111** [comment]
Ok, a web based backup interface it is! I think this can and will be done but it proably wont be the biggest priority since there are many good options that already work pretty well on the command line. 

In the meantime you can try this script i personally use for my backups https://github.com/vabene1111/DockerPostgresBackups but be aware that its not really polished/edge case proof

**Comment by vabene1111** [comment]
while django-dbbackup in combination with https://github.com/timonweb/django-dbbackup-ui looks really good it appears to be full of bugs where people are having all kinds of problems different databases, special characters in the DB and storage backends. 

Need to find a better solution as reliability is the number one priority when it comes to backups ...

**Comment by vabene1111** [comment]
ok so i have decided to implement backups using djangos `dumpdata` and `loaddata` commands. 

- [ ] include mediafiles in backup
- [ ] compress backup
- [ ] non super user visibility
- [ ] restore database
- [ ] restore media files

**Comment by thickconfusion** [comment]
Just wanted to drop by and say this thread was helpful for the following:
I just migrated Tandoor from an armv71 docker host to an arm64 docker host. In general it's easy to move bind and volume mounts by carefully stopping the containers, gzipping their volumes, moving to the new host, and firing back up.

However, encountered a strange issue. Because of the architecture change, Postgre (db_recipes) was giving me a strange error: 
```
FATAL:  database files are incompatible with server
DETAIL:  The database cluster was initialized without USE_FLOAT8_BYVAL but the server was compiled with USE_FLOAT8_BYVAL.
HINT:  It looks like you need to recompile or initdb.
LOG:  database system is shut down
```
So I ended up using a form of this https://github.com/vabene1111/DockerPostgresBackups to create a dump, then restore it on the new machine. Voila!

**Comment by vabene1111** [comment]
awesome, database migration is always easiest with dumps because it is independet of the version and environment and even if it is you can just change text to get it working. Glad it helped

**Comment by Mikhail5555** [comment]
Sorry for bumping this old issue, but would a `python manage.py dumpdata` followed by a `python manage.py loaddata` work when you want to migrate from a sqlite database to a postgresql database (or is there better approach)?

**Comment by vabene1111** [comment]
that is likely one of the best appraoches but make sure to properly test it before deleting the old DB. There are several foreign relations across the database and loaddata might have a hard time handeling those. 

I know there is an old discussion from someone who did exactly what you are planning to do, maybe that helps you.

**Comment by Szeraax** [comment]
I just successfully exported all my recipes to a zip file via web interface and was able to import them. Is there still any work to be done on this issue? Or are we good to close it?

**Comment by vabene1111** [comment]
yes, the recipes are not a real backup because there are supermarkets, food metadata, autoamtions an much more that should also be included in a real backup. This is planned at some point

**Comment by gruhland** [comment]
Sorry to add a new comment to this thread. I just updated my docker installation. Then I tried to export ALL recipes. The information window shows the progress of the export but I get no automatic download. The text on the page shows immediately "If download did not start automatically: Herunterladen (Expired)" where download (Herunterladen) is crossed out. If I try to export only one recipe it is no problem. Any idea?
Regards
Goetz

**Comment by smilerz** [comment]
> Sorry to add a new comment to this thread. I just updated my docker installation. Then I tried to export ALL recipes. The information window shows the progress of the export but I get no automatic download. The text on the page shows immediately "If download did not start automatically: Herunterladen (Expired)" where download (Herunterladen) is crossed out. If I try to export only one recipe it is no problem. Any idea? Regards Goetz

it should be fixed in the next release - in the meantime, if you hit refresh it will frequently start the download.

**Comment by gruhland** [comment]
Thanks, I will try :-)

**Comment by bennysp** [comment]
Here is what I did to migrate my Tandoor.

## Current Host (from TrueNAS Docker):
`pg_dump -U recipes -h localhost -d db_recipes -Fc -f tandoor_db.dump`
`tar -czvf tandoor_media.tar.gz /opt/recipes/mediafiles`

Copied these to the new host.

## New Host (Proxmox LXC via Helper Scripts):
In this case, because the helper scripts create the install and data, I had to dump my data first... and then recreate...

Stop Tandoor

### 1) Become postgres OS user (inside DB container/host)
`su - postgres`

### 2) Terminate open sessions and recreate DB
```shell
psql -d postgres -c "SELECT pg_terminate_backend(pid) FROM pg_stat_activity WHERE datname='db_recipes' AND pid <> pg_backend_pid();"
dropdb db_recipes
createdb -O tandoor db_recipes
psql -d db_recipes -c "CREATE EXTENSION IF NOT EXISTS unaccent;"
psql -d db_recipes -c "CREATE EXTENSION IF NOT EXISTS pg_trgm;"
exit
```

### 3) Restore
`pg_restore -U tandoor -h localhost -d db_recipes --no-owner --no-privileges --no-comments tandoor_db.dump`

Start Tandoor

I haven't noticed any issues.

**Comment by gruhland** [comment]
Thanks for the explanation. I will give it a try, when I have some time.
Best regards
Goetz

05.09.2026 01:45:04 Ben Ptacek ***@***.***>:
